### PR TITLE
Go Modules & Makefiles

### PR DESCRIPTION
**Pull request by gavinbunney**
- Update to using go-modules
- Added Makefile & scripts for building distributions of plugin

### CODE REVIEW

**Comment by gavinbunney** [comment]
@timperrett This also needs a travis-ci API key for the deployments to work 🙏
Placeholder with `TODO-GET KEY`

**Comment by timperrett** [comment]
@gavinbunney thanks for this - appear to be some merge conflicts?

**Comment by gavinbunney** [comment]
I'll fix... looks like @drewgonzales360 PRs snuck in a broken things 😛

**Comment by gavinbunney** [comment]
@timperrett Fixed, squashed and ready to go

File: .travis.yml [hunk 1]
```diff
@@ -1,34 +1,38 @@
 language: go
+sudo: required
```
Comment (timperrett) [inline .travis.yml]: This would make it run on a dedicated VM, which i dont believe is needed for such a small project. Can we remove?

**Comment by timperrett** [comment]
Just that one small comment - otherwise looks great @gavinbunney

File: .travis.yml (hunk 1 above)
Comment (gavinbunney) [reply .travis.yml]: Fair point - copy and paste :D Removed!

**Comment by gavinbunney** [comment]
Once merged, could you also add the api-key for deployments @timperrett ? Thanks 👏

**Comment by gavinbunney** [comment]
@timperrett Are you able to merge this PR and add api-keys? I would like to download this plugin directly from github without having to worry about locally compiling and uploading elsewhere ;)

**Pull request merged**